app/src/main/java/org/p2p/wallet/swap/jupiter/statemanager/token_selector/CommonSwapTokenSelector.kt

```diff
@@ -0,0 +1,66 @@
+package org.p2p.wallet.swap.jupiter.statemanager.token_selector
+
+import org.p2p.core.utils.Constants
+import org.p2p.wallet.home.repository.HomeLocalRepository
+import org.p2p.wallet.infrastructure.dispatchers.CoroutineDispatchers
+import org.p2p.wallet.swap.jupiter.domain.model.SwapTokenModel
+import org.p2p.wallet.swap.jupiter.repository.tokens.JupiterSwapTokensRepository
+import kotlinx.coroutines.async
+import kotlinx.coroutines.withContext
+
+class CommonSwapTokenSelector(
+    private val jupiterTokensRepository: JupiterSwapTokensRepository,
+    private val homeLocalRepository: HomeLocalRepository,
+    private val dispatchers: CoroutineDispatchers,
+) : InitialTokenSelector {
+
+    override suspend fun getTokenPair(): Pair<SwapTokenModel, SwapTokenModel> = withContext(dispatchers.io) {
+        val jupiterTokensJob = async { jupiterTokensRepository.getTokens() }
+        val userTokensJob = async { homeLocalRepository.getUserTokens() }
+        val jupiterTokens = jupiterTokensJob.await()
+        val userTokens = userTokensJob.await()
+        var haveUCDC = false
```

Comment:
userHasUsdc = userTokens.any { it.isUSDC } 
userHasSol = userTokens.any { it.isSOL }

Reply:
double loop, but ok